src/App.tsx

```diff
@@ -1,27 +1,82 @@
-import React from 'react';
+import React, { useState } from 'react';
+import Loader from 'react-loaders';
 import './App.scss';
 import { GoodsList } from './GoodsList';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+import { getAll, getFirst5, getRedGoods } from './api/goods';
+import { Good } from './types/Good';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+export const App: React.FC = () => {
+  const [visibleGoods, setVisibleGoods] = useState<Good[]>([]);
+  const [hasLoadingError, setHasLoadingError] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+  const fetchData = async (getData: { (): Promise<Good[]>; }) => {
```

Comment:
Check if there is no error and the same data is requested to avoid re-fetching the same data multiple times.